.env

```diff
@@ -3,3 +3,4 @@ XPUB_VERSION=d112a742237ba8b92b8522e55157fd77cf075e26
 ORCID_CLIENT_ID=fake_client_id
 ORCID_CLIENT_SECRET=fake_client_secret
 PUBSWEET_BASEURL=http://localhost:3000
+PUBSWEET_HOSTNAME=localhost
```

Comment:
Is this used internally or exposed to the browser? If the latter, we need to add it https://github.com/elifesciences/elife-xpub-formula/blob/master/salt/elife-xpub/init.sls#L26-L28 which is the server-based deployment rather than the local one.

Reply:
I would assume the http server where this is used has access to these environment variables. Was this your question? How can I test if the container has the variables?

Reply:
The only place where these variables should be available is by being substituted in https://github.com/elifesciences/elife-xpub-deployment/blob/develop/docker-compose.yml (see the `ORCID_*` ones for example). If you need to inject them into any container, they have to be added explicitly there (see `NODE_ENV` for example). I notice now this is missing.

The server will then be able to override this file with environment variables, by adding a value in [this list](https://github.com/elifesciences/elife-xpub-formula/blob/master/salt/elife-xpub/init.sls#L26-L28). But if this if `localhost` everywhere no need to do that part.

Reply:
Should I make a new PR trying to add `PUBSWEET_HOSTNAME` to that `docker-compose.yml` file? It just needs to be in env on the same container as the server for it to work https://github.com/elifesciences/elife-xpub/blob/4b93c366efed7e9d6fcc0d10e646163379f069ee/config/custom-environment-variables.js

Reply:
From your link I understand you are trying to add it to `pubsweet-server.hostname`. Then yes, add a line [here](https://github.com/elifesciences/elife-xpub-deployment/blob/develop/docker-compose.yml#L22) mapping the environment variable of docker-compose into the JSON structure for `NODE_CONFIG`. I say this just to be consistent with the rest of the configurations which are all inserted into `NODE_CONFIG`.

(incidentally, this won't make PUBSWEET_HOSTNAME appear as an environment variable for the containers, so if that is required for another reason then this solution is no longer valid)

Reply:
Sorry for being confusing, in that link I posted node/the server will try to read the value of `PUBSWEET_HOSTNAME` from environment and store it in `pubsweet-server.hostname`. If the server doesn't have access to this `PUBSWEET_HOSTNAME` env variable then that line has no effect and I will remove it. I initially thought that this PR will add PUBSWEET_HOSTNAME to env in the same container as the server.

I will make a new PR to add hostname to pubsweet-server in that file you linked me.